File: client/src/types/booking.ts
export type BookingStatus = 'pending' | 'confirmed' | 'cancelled' | 'completed';

export interface Booking {
  id: number;
  booking_number: string;
  property_id: number;
  property_name?: string;
  property_locality?: string;
  property_city?: string;
  property_image?: string;
  property_price?: number;
  status: BookingStatus;
  customer_name: string;
  customer_email: string;
  customer_phone: string;
  preferred_visit_date?: string;
  visit_time_slot?: string;
  special_requirements?: string;
  created_at: string;
  updated_at?: string;
}

export interface BookingCreateData {
  property_id: number;
  customer_name: string;
  customer_email: string;
  customer_phone: string;
  customer_address?: string;
  pan_number?: string;
  aadhaar_number?: string;
  preferred_visit_date?: string;
  visit_time_slot?: string;
  special_requirements?: string;
  documents?: Array<{
    document_type: string;
    title: string;
    file_name: string;
    file_url: string;
    mime_type?: string;
    file_size_bytes?: number;
  }>;
}


export interface WishlistItem {
  id: number;
  property_id: number;
  property_name?: string;
  property_locality?: string;
  property_city?: string;
  property_price?: number;
  property_image?: string;
  property_bedrooms?: number;
  property_bathrooms?: number;
  property_area_sqft?: number;
  property_status?: string;
  property_builder?: string;
  property_slug?: string;
  property_rating?: number;
  created_at: string;
}

export interface Review {
  id: number;
  property_id?: number;
  rating: number;
  title?: string;
  comment?: string;
  reviewer_name?: string;
  reviewer_avatar?: string;
  is_verified: boolean;
  created_at: string;
}

File: server/schemas/booking.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class BookingCreate(BaseModel):
    property_id: int
    customer_name: str
    customer_email: str
    customer_phone: str
    customer_address: Optional[str] = None
    pan_number: Optional[str] = None
    aadhaar_number: Optional[str] = None
    preferred_visit_date: Optional[datetime] = None
    visit_time_slot: Optional[str] = None
    special_requirements: Optional[str] = None
    documents: Optional[list[dict]] = None



class BookingUpdate(BaseModel):
    status: Optional[str] = None
    preferred_visit_date: Optional[datetime] = None
    visit_time_slot: Optional[str] = None
    special_requirements: Optional[str] = None
    agent_notes: Optional[str] = None


class BookingOut(BaseModel):
    id: int
    booking_number: str
    property_id: int
    property_name: Optional[str] = None
    property_locality: Optional[str] = None
    property_city: Optional[str] = None
    property_image: Optional[str] = None
    property_price: Optional[int] = None
    status: str
    customer_name: str
    customer_email: str
    customer_phone: str
    preferred_visit_date: Optional[datetime] = None
    visit_time_slot: Optional[str] = None
    special_requirements: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}
